rex/resource.py

```diff
@@ -1253,11 +1254,130 @@ def _get_ds(self, ds_name, ds_slice):
+               "index, or use a 1-dimensional slice."
+               .format(ds_name, ds.shape, meta_shape, ti_shape))
+        raise ResourceRuntimeError(msg)
+
+    def _get_ds_with_spatial_repeat(self, ds, ds_name, ds_slice):
+        """
+        Extract 1D data using 2D slice by repeating the 1D data along
+        the spatial dimension
+
+        Parameters
+        ----------
+        ds : h5py.dataset
+            Open .h5 dataset instance to extract data from
+        ds_name : str
+            Variable dataset to be extracted
+        ds_slice : tuple
+            Tuple of (int, slice, list, ndarray) of what to extract from ds,
+            each arg is for a sequential axis
+
+        Returns
+        -------
+        out : ndarray
+            ndarray of variable timeseries data
+            If unscale, returned in native units else in scaled units
+        """
+        msg = ("Attempting to use a 2D slice on a 1D dataset ({!r}). 1D "
+               "dataset values will be repeated along the spatial dimension"
+               .format(ds_name))
+        warn(msg)
+
+        out = ResourceDataset.extract(ds, ds_slice[:1],
+                                      scale_attr=self.SCALE_ATTR,
                                       add_attr=self.ADD_ATTR,
                                       unscale=self._unscale)
+        try:
+            out = (np.ones((len(out), self.shapes['meta'][0])).T * out).T
+            return out[:, ds_slice[1]]
+        except TypeError:
+            out = np.ones(self.shapes['meta']) * out
+            return out[ds_slice[1]]
 
-        return out
+    def _get_ds_with_temporal_repeat(self, ds, ds_name, ds_slice):
+        """
+        Extract 1D data using 2D slice by repeating the 1D data along
+        the temporal dimension
+
+        Parameters
+        ----------
+        ds : h5py.dataset
+            Open .h5 dataset instance to extract data from
+        ds_name : str
+            Variable dataset to be extracted
+        ds_slice : tuple
+            Tuple of (int, slice, list, ndarray) of what to extract from ds,
+            each arg is for a sequential axis
+
+        Returns
+        -------
+        out : ndarray
+            ndarray of variable timeseries data
+            If unscale, returned in native units else in scaled units
+        """
+        msg = ("Attempting to use a 2D slice on a 1D dataset ({!r}). 1D "
+               "dataset values will be repeated along the temporal dimension"
+               .format(ds_name))
+        warn(msg)
+
+        out = ResourceDataset.extract(ds, ds_slice[1:],
+                                      scale_attr=self.SCALE_ATTR,
+                                      add_attr=self.ADD_ATTR,
+                                      unscale=self._unscale)
+        try:
+            out = np.ones((self.shapes['time_index'][0], len(out))) * out
+        except TypeError:
+            out = np.ones(self.shapes['time_index']) * out
```

Comment:
you should convert the returned data to float32. The np.ones() will make everything into float64 (unnecessary precision and a waste of big mem with 5min data)

Reply:
Very good catch, thank you!!